### PR TITLE
Feature/pdct 1661 world map should only show counts of families from allowed corpora 

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

- Update world map query to respect allowed corpora
- Return 400 if allowed corpora is None or []

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Refactor legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] **DB_CLIENT DEPENDENCY IS ON THE LATEST VERSION**
- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

**Review by annaCPR — CHANGES_REQUESTED** [review]

File: app/repository/sql/world_map.sql
Comment (jamesgorrie) [inline app/repository/sql/world_map.sql:1]: When do we use SQL over python to get data?

File: tests/non_search/routers/geographies/test_world_map_summary.py [hunk 1]
```diff
@@ -122,27 +121,89 @@ def test_endpoint_returns_ok_some_docs_per_published_family_unpublished(
...
+        .filter(Family.family_status == FamilyStatus.PUBLISHED)
+        .filter(Geography.display_value == geo_display_value)
+        .join(FamilyGeography, Family.import_id == FamilyGeography.family_import_id)
+        .join(FamilyCorpus, Family.import_id == FamilyCorpus.family_import_id)
+        .join(Corpus, Corpus.import_id == FamilyCorpus.corpus_import_id)
+        .join(Geography, Geography.id == FamilyGeography.geography_id)
+        .filter(Corpus.import_id == "UNFCCC.corpus.i00000001.n0000")
+        .all()
+    )
+    print(fams)
```
[the remark this answers is not in the document]
Comment (katybaulch) [reply tests/non_search/routers/geographies/test_world_map_summary.py:198]: You definitely will when things go wrong 😅 I'll take it out though, as I trust we all know how to add it back in haha

File: tests/non_search/routers/geographies/test_world_map_summary.py [hunk 2]
```diff
@@ -122,27 +121,89 @@ def test_endpoint_returns_ok_some_docs_per_published_family_unpublished(
...
+def test_endpoint_returns_different_results_with_alt_token(
+    data_db,
+    data_client,
+    geo_display_value,
+    expected_exec,
+    expected_leg,
+    expected_unfccc,
+    alternative_token,
+):
+    """Check endpoint returns 200 & only counts CCLW docs"""
```
[the remark this answers is not in the document]
Comment (katybaulch) [reply tests/non_search/routers/geographies/test_world_map_summary.py:168]: Updated, good spot.

File: app/repository/sql/world_map.sql
Comment (katybaulch) [reply app/repository/sql/world_map.sql:1]: When the query becomes disgusting to maintain or too complex.

File: app/api/api_v1/routers/world_map.py
[the remark this answers is not in the document]
Comment (katybaulch) [reply app/api/api_v1/routers/world_map.py:1]: For now, I will keep the world map bits in here until we've had that discussion - as it's easier from a maintainability perspective to understand what this file is for from the naming - whereas previously it was confusing whether it was to do with the geography page (which this endpoint has nothing to do with) or the world map (which it does have something to do with).

File: tests/non_search/routers/geographies/test_world_map_summary.py [hunk 3]
```diff
@@ -95,20 +96,18 @@ def test_endpoint_returns_ok_some_docs_per_published_family_unpublished(
     expected_exec,
     expected_leg,
     expected_unfccc,
+    valid_token,
 ):
     """Check endpoint returns 200 & discounts CREATED & DELETED docs"""
```
[the remark this answers is not in the document]
Comment (katybaulch) [reply tests/non_search/routers/geographies/test_world_map_summary.py:101]: This endpoint does exactly what it says in the docstring -- the helper function _make_world_map_lookup_request does this assert on the status code.

File: tests/non_search/routers/geographies/test_world_map_summary.py [hunk 4]
```diff
@@ -45,20 +48,18 @@ def test_endpoint_returns_ok_all_docs_per_family_published(
     expected_exec,
     expected_leg,
     expected_unfccc,
+    valid_token,
 ):
     """Check endpoint returns 200 on success"""
```
[the remark this answers is not in the document]
Comment (katybaulch) [reply tests/non_search/routers/geographies/test_world_map_summary.py:53]: The helper function _make_world_map_lookup_request does this assert on the status code. I'll make the docstring a little more descriptive though.

[the remark this answers is not in the document]
Comment (annaCPR) [reply tests/non_search/routers/geographies/test_world_map_summary.py]: Cool, thanks.

**Pull request merged**